src/scripts/scaling/scalingAction.sh

```diff
@@ -16,10 +14,291 @@ scaling_size=1
 access_token=""
 kubernetes_master="https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}"
 
+# timestamp
+#   purpose:  echo timestamp in the form yyyymmddThh:mm:ss.mmm ZZZ
+#   example:  20181001T14:00:00.001 UTC
+function timestamp() {
+  local timestamp="`date --utc '+%Y-%m-%dT%H:%M:%S %N %s %Z' 2>&1`"
+  if [ ! "${timestamp/illegal/xyz}" = "${timestamp}" ]; then
+    # old shell versions don't support %N or --utc
+    timestamp="`date -u '+%Y-%m-%dT%H:%M:%S 000000 %s %Z' 2>&1`"
+  fi
+  local ymdhms="`echo $timestamp | awk '{ print $1 }'`"
+  # convert nano to milli
+  local milli="`echo $timestamp | awk '{ print $2 }' | sed 's/\(^...\).*/\1/'`"
+  local secs_since_epoch="`echo $timestamp | awk '{ print $3 }'`"
+  local millis_since_opoch="${secs_since_epoch}${milli}"
+  local timezone="`echo $timestamp | awk '{ print $4 }'`"
+  echo "${ymdhms}.${milli} ${timezone}"
+}
+
+function trace() {
+  echo "@[$(timestamp)][INFO]" "$@"
```

Comment:
It looks like every single trace is redirected to 'scalingAction.log', so the redirect can be centralized into this function.

Also, can the same log file be shared by multiple different clusters and domains?  If so, then it'd be helpful to decorate each trace with the current namespace, domain-uid, and cluster-name "[ns=xxx][uid=xxx][cl=xxx]".

Reply:
Great suggestion.  Updated the trace logging to include the above suggestions

Reply:
LGTM